indexing-service/src/main/java/io/druid/indexing/common/task/HadoopTask.java

```diff
@@ -69,6 +74,24 @@ protected HadoopTask(
     return hadoopDependencyCoordinates == null ? null : ImmutableList.copyOf(hadoopDependencyCoordinates);
   }
 
+  protected static final Predicate<URL> IS_DRUID_URL = new Predicate<URL>()
+  {
+    @Override
+    public boolean apply(@Nullable URL input)
+    {
+      try {
+        if (input == null) {
+          return false;
+        }
+        final String fName = Paths.get(input.toURI()).getFileName().toString();
+        return fName.startsWith("druid") && fName.endsWith(".jar");
```

Comment:
is there any way this can be more robust?

I'm imagining one day someone having a bad time because their extension depends on a jar called "druid-mything-adapter.jar"